### PR TITLE
Add header validation to MockTeapot if headers are provided

### PR DESCRIPTION
**Pull request by designatednerd**
This preserves existing behavior in that if no headers are provided, the check just returns true. 

However, if you provide headers to check for, it validates that all of your provided headers are there before returning. It does not validate exhaustively: Other headers may be present and you will not receive an error. It is only checking for the specific headers you are sending. 

This can help catch bugs with POST and/or PUT signatures which need to be validated in tests without having to hit a live API.

### CODE REVIEW

File: Teapot/MockTeapot.swift [hunk 1]
```diff
@@ -95,4 +117,20 @@ open class MockTeapot: Teapot {
             completion(nil, TeapotError.missingMockFile(resource))
         }
     }
+
+    private func checkHeadersAgainstExpected(headers: [String: String]?) -> Bool {
+        guard !headersToCheckFor.isEmpty else { /* nothing to check */ return true }
+
+        guard let received = headers else { /* We want to check but there's nothing to check */ return false }
```
Comment (marijnschilling) [inline Teapot/MockTeapot.swift]: This comment should be something like `Headers expected but not received` to be more clear

Comment (marijnschilling) [reply Teapot/MockTeapot.swift]: Oh and i think `receivedHeaders` as a var would make this easier to read.

File: Teapot/MockTeapot.swift [hunk 2]
```diff
@@ -95,4 +117,20 @@ open class MockTeapot: Teapot {
...
+
+        for (key, value) in headersToCheckFor {
```
Comment (marijnschilling) [inline Teapot/MockTeapot.swift]: Can't there be a function that just checks wether `receivedHeaders.containts(headersToCheckFor)`?

**Review by marijnschilling — APPROVED** [review]

File: Teapot/MockTeapot.swift (hunk 2 above)
Comment (designatednerd) [reply Teapot/MockTeapot.swift]: There's only a `contains(where:)` on dictionaries - I don't think it'd actually be that much shorter than this, and it probably wouldn't be as clear

File: Teapot/MockTeapot.swift (hunk 1 above)
Comment (elland) [reply Teapot/MockTeapot.swift]: not a fan of inlined comments inside code like this. makes it harder to figure out where they start/end without syntax highlight.

Comment (designatednerd) [reply Teapot/MockTeapot.swift]: Fair enough, was trying to keep to the preference for one-line return statements. I will break these out.

**Comment by elland** [comment]
Looks fine to me. 👍🏽

**Comment by designatednerd** [comment]
@Elland @marijnschilling updated the cancel request test to make it less flaky as well.

**Review by elland — APPROVED** [review]
Looks good to me. 👍

**Pull request merged**